### PR TITLE
add extract method

### PR DESCRIPTION
**Pull request by AntoniosMalak**

### CODE REVIEW

**Comment by AntoniosMalak** [comment]
Hi, @simonetgordon may u review this, please?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by simonetgordon** [comment]
@AntoniosMalak, apologies for the delay in reviewing, I was on Christmas break.

**Comment by AntoniosMalak** [comment]
Merry Christmas @simonetgordon, Now it was finished and I see the errors in another code like test-tensorflow and there is no error in test-numpy. So what should I do next?

**Comment by AntoniosMalak** [comment]
Hi `@simonetgordon` may u check it

File: ivy/functional/frontends/numpy/sorting_searching_counting/searching.py [hunk 1]
```diff
@@ -83,7 +83,5 @@ def nanargmax(a, /, *, axis=None, out=None, keepdims=False):
 
 
 @to_ivy_arrays_and_back
-@from_zero_dim_arrays_to_scalar
```
Comment (simonetgordon) [inline ivy/functional/frontends/numpy/sorting_searching_counting/searching.py:86]: Can you reinstate this function please? It is causing a key error in the test collection which is preventing `test_numpy_extract` from running. Thanks!

**Review by simonetgordon — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/numpy/sorting_searching_counting/searching.py (hunk 1 above)
Comment (AntoniosMalak) [reply ivy/functional/frontends/numpy/sorting_searching_counting/searching.py:86]: Done..

File: ivy/functional/frontends/numpy/sorting_searching_counting/searching.py [hunk 2]
```diff
@@ -84,6 +84,5 @@ def nanargmax(a, /, *, axis=None, out=None, keepdims=False):
 
 @to_ivy_arrays_and_back
 @from_zero_dim_arrays_to_scalar
-def nanargmin(a, /, *, axis=None, out=None, keepdims=False):
```
Comment (simonetgordon) [inline ivy/functional/frontends/numpy/sorting_searching_counting/searching.py:87]: `nanargmin` is still missing from your code. If you look at the check PR `numpy-frontends-test -> Details -> Run Frontend Tests`, you'll see that it's not running due to a KeyError in the corresponding test function (can't find the function to test)
```
==================================== ERRORS ====================================
_ ERROR collecting ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py _
ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:373: in <module>
    def test_numpy_nanargmin(
ivy_tests/test_ivy/helpers/testing_helpers.py:251: in test_wrapper
    callable_fn, fn_name, fn_mod = _import_fn(fn_tree)
ivy_tests/test_ivy/helpers/testing_helpers.py:131: in _import_fn
    callable_fn = mod.__dict__[fn_name]
E   KeyError: 'nanargmin'
```

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Review by simonetgordon — APPROVED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 3]
```diff
@@ -383,3 +395,32 @@ def test_numpy_nanargmin(
...
+    broadcastables,
+    as_variable,
+    num_positional_args,
+    native_array,
+    frontend,
+    fn_tree,
+    on_device,
+):
+    a, cond, dtype = broadcastables
+    helpers.test_frontend_function(
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:415]: Currently the test is failing with the error:
```
=================================== FAILURES ===================================
______ test_numpy_extract[cpu-ivy.functional.backends.torch-False-False] _______
ivy/exceptions.py:118: in new_fn
    return fn(*args, **kwargs)
ivy/functional/backends/torch/data_type.py:149: in as_ivy_dtype
    return ivy.Dtype(ivy_dtype_dict[dtype_in])
E   TypeError: unhashable type: 'list'
```

See the test output of the first check running on this PR for the full stack trace (`run_tests_1`) for more details and the decorator to add to the test locally to reproduce the failure if you are currently not seeing it.

**Review by simonetgordon — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 4]
```diff
@@ -383,3 +395,32 @@ def test_numpy_nanargmin(
...
+    as_variable,
+    num_positional_args,
+    native_array,
+    frontend,
+    fn_tree,
+    on_device,
+):
+    cond, a, dtype = broadcastables
+    helpers.test_frontend_function(
+        input_dtypes=[dtype, "bool"],
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:416]: change `input_dtypes` from list to a tuple

**Review by simonetgordon — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 5]
```diff
@@ -383,3 +395,32 @@ def test_numpy_nanargmin(
...
         keepdims=keep_dims,
     )
+
+    
+# extract
+@handle_frontend_test(
+    fn_tree="numpy.extract",
+    broadcastables=_broadcastable_duo(),
+)
+def test_numpy_extract(
```
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:405]: If you check out the intelligent-tests-1 error stack trace, you'll see some errors like
```
ivy/func_wrapper.py:191: in new_fn
    ret = fn(*args, **kwargs)
ivy/functional/backends/numpy/data_type.py:105: in astype
    return x.astype(dtype)
E   TypeError: Field elements must be 2- or 3-tuples, got ''int64''
```
Try re-generating the example which caused this to try and get to them bottom of the issue. This is done by adding the decorator printed in the error stack trace to the `test_numpy_extract` locally

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**